airflow/auth/managers/base_auth_manager.py

```diff
@@ -42,6 +43,16 @@ def get_user_name(self) -> str:
         """Return the username associated to the user in session."""
         ...
 
+    @abstractmethod
+    def get_user(self) -> BaseUser:
+        """Return the user associated to the user in session."""
+        ...
+
+    @abstractmethod
+    def get_user_id(self) -> str:
+        """Return the user ID associated to the user in session."""
+        ...
```

Comment:
I think the ID is an int? At least that’s the case for Flask’s user model.

Reply:
Yeah but we cannot assert it will be the case for all auth managers. Since forecasting is kind of hard, I rather have it as a string

Reply:
Then do we want to do a `str()` cast in the subclass so it actually returns a str? Right now I think the returned value would be an int and not match the annotation.

Reply:
Sounds good